src/components/transport_manager/src/cloud/cloud_websocket_connection_factory.cc

```diff
@@ -56,11 +56,15 @@ TransportAdapter::Error CloudWebsocketConnectionFactory::Init() {
 TransportAdapter::Error CloudWebsocketConnectionFactory::CreateConnection(
     const DeviceUID& device_uid, const ApplicationHandle& app_handle) {
   LOG4CXX_AUTO_TRACE(logger_);
-  std::shared_ptr<WebsocketClientConnection> connection =
-      std::make_shared<WebsocketClientConnection>(
-          device_uid, app_handle, controller_);
-  controller_->ConnectionCreated(connection, device_uid, app_handle);
-  TransportAdapter::Error error = connection->Start();
+  auto connection = controller_->FindPendingConnection(device_uid, app_handle);
+
+  std::shared_ptr<WebsocketClientConnection> ws_connection =
+      std::dynamic_pointer_cast<WebsocketClientConnection>(connection);
+  if (!ws_connection) {
```

Comment:
Seems that the safer way to check whether the shared pointer is empty is `ptr.use_count() == 0`, since it accounts for the case where the pointer does not manage any objects, but the stored pointer is not `null`.